File: api/src/main/java/whenyourcar/api/web/controller/chat/ChatBrokerController.java
package whenyourcar.api.web.controller.chat;

import lombok.RequiredArgsConstructor;
import org.springframework.messaging.handler.annotation.DestinationVariable;
import org.springframework.messaging.handler.annotation.MessageMapping;
import org.springframework.stereotype.Controller;
import whenyourcar.application.facade.chat.ChatBrokerFacade;

@Controller
@RequiredArgsConstructor
public class ChatBrokerController {
    private final ChatBrokerFacade chatBrokerFacade;
    @MessageMapping("/{roomId}/{sender}")
    public void postMessageToRedisStream(@DestinationVariable Long roomId,
                                         @DestinationVariable Long sender,
                                         String message) {
        chatBrokerFacade.sendMessageToQueue(roomId, sender, message);
    }
}

File: api/src/main/kotlin/howmuchcar/api/controller/chat/ChatBrokerController.kt
package howmuchcar.api.controller.chat

import howmuchcar.application.facade.chat.ChatMessageFacade
import howmuchcar.domain.entity.User
import lombok.RequiredArgsConstructor
import org.springframework.messaging.handler.annotation.DestinationVariable
import org.springframework.messaging.handler.annotation.MessageMapping
import org.springframework.stereotype.Controller

@Controller
@RequiredArgsConstructor
class ChatBrokerController(
    private val chatMessageFacade: ChatMessageFacade
) {
    @MessageMapping("/{roomId}/{senderId}")
    fun postMessageToRedisStream(@DestinationVariable roomId: Long,
                                 @DestinationVariable senderId: Long,
                                 message: String ) {
        chatMessageFacade.sendMessageToQueue(roomId, senderId, message)
    }
}